qiskit/quantum_info/operators/measures.py

```diff
@@ -350,6 +350,94 @@ def cvx_bmat(mat_r, mat_i):
     return sol
 
 
+def unitary_diamond_distance(op1: Operator, op2: Operator) -> float:
```

Comment:
An idea is this could be implemented as a general function
```suggestion
def diamond_distance(op1: BaseOperator, op2: BaseOperator) -> float:
```
where if the operator type are unitary you can have special cases for implementing it eg this current case for Operators, and in general this can call diamond norm for `op1 - op2` appropriately normalized.

There is also an efficient expression for the diamond distance between Pauli channels that could also be added as a special case later as well.

Probably the trick used here could also be special cased for Pauli's and Cliffords to have an efficient version for those that works for large numbers of qubits?